tests/unit/lms/services/grouping_test.py

```diff
@@ -0,0 +1,69 @@
+from unittest.mock import sentinel
+
+import pytest
+
+from lms.models import Grouping
+from lms.models._hashed_id import hashed_id
+from lms.services.grouping import GroupingService, factory
+from tests import factories
+
+pytestmark = pytest.mark.usefixtures("course_service", "application_instance_service")
+
+
+class TestGroupingService:  # pylint: disable=protected-access
+    CONTEXT_ID = "context_id"
+    TOOL_CONSUMER_INSTANCE_GUID = "t_c_i_guid"
+
+    def test_upsert_inserts(self, svc):
+        # Start with no groupings
+        assert not svc._db.query(Grouping).count()
+
+        svc.upsert(
+            Grouping(
+                application_instance=factories.ApplicationInstance(),
+                authority_provided_id="ID",
+                lms_id="lms_id",
+                lms_name="lms_name",
+            )
+        )
+        assert svc._db.query(Grouping).count() == 1
+
+    def test_upsert_updates(self, svc):
+        grouping = factories.Grouping()
+        svc._db.flush()
+
+        grouping.lms_name = "new_name"
+
+        grouping = svc.upsert(grouping)
+
+        db_grouping = svc._db.query(Grouping).one()
+        assert db_grouping.lms_name == "new_name"
+
+    def test_section_grouping_finds_course(
```

Comment:
The test should be named after the method that it tests:

```suggestion
    def test_canvas_section_finds_course(
```